bibtex-actions.el

```diff
@@ -123,7 +133,7 @@ candidate list"
                        (affixation-function . bibtex-actions--affixation)
                        (category . bibtex))
                    (complete-with-action action candidates string predicate)))
-                 nil nil initial nil nil nil)))
+                 nil nil initial 'bibtex-actions-history 'bibtex-actions-presets nil)))
```

Comment:
> I keep getting this `Wrong type argument: char-or-string-p` error when calling `bibtex-actions-insert-citation` and others, whenever I try to navigate through the history

Yeah, I was just going to report the same. Maybe there's some little lispy thing wrong with this line, though I don't see what.

I thought maybe the type on the defcustom is wrong, but I don't believe it is.

Reply:
> Just remove the quote from `'bibtex-actions-presets`, you need to pass a list of string, not a symbol naming a variable containing the list. (It's different for the history, there you need to pass the name of the history variable, so completing read can modify it.)

Thank you for the explanation; those little details keep tripping me up.

Fix pushed, and now works.

I only tested it really briefly, but first impression is I like it!

Let me know WYT @apc.

I guess the only remaining for question for me beyond the general UX one is whether to adapt that `bibtex-actions-insert-preset` function to this (it won't be persistent ATM; right), or just keep it simple for now and remove it, and tell users to add it to their config manually.

Reply:
> (it won't be persistent ATM; right)

What do you mean?

Reply:
The history variable I know/assume will get saved by savehist.

Is that also true of the future history? I was assuming not for some reason.

Reply:
The future history is sort of created fresh for every call to `completing-read`, by whatever expression you, the developer, use as the DEF argument. Here you simply pass the value of the variable `bibtex-actions-presets`, so it's a matter of how that variable gets its value. I was assuming that users wouldn't need this persisted since they would actually just assign that variable a value in their init file, a value consisting of the predefined searches they find useful.

Reply:
> I was assuming that users wouldn't need this persisted since they would actually just assign that variable a value in their init file, a value consisting of the predefined searches they find useful.

Yes, sorry; that's what I meant.

In that case, there's no point for that function; right?

Reply:
I don't use savehist, so I can say from firsthand experience, but the docstring seems to suggest users need to add `'bibtex-actions-history` to `savehist-additional-variables` if they want it saved.

Reply:
(Why don't I use savehist?)

Reply:
> Yes, sorry; that's what I meant. 
> In that case, there's no point for that function; right?

You mean the `bibtex-actions-insert-preset` function? I think it depends on how many predefined searches users will have. The future history mechanism only lets you scroll through presets one at a time. If a user has lot of presets, they'd probably prefer a function like `bibtex-actions-insert-preset`. (Note that a command like `consult-history` only completes items from the actual history, not from the so-called "future history".)

Reply:
> The future history mechanism only lets you scroll through presets one at a time. If a user has lot of presets ...

These two are in some tension though.

Seems you'd want to have less than 10 for it be worth using.

I'm leaning towards removing it, but will sleep on it.

Reply:
I decided to keep. Can see what feedback comes in.